src/Datadog.Trace.ClrProfiler.Native/clr_helpers.h

```diff
@@ -52,16 +55,23 @@ class EnumeratorIterator {
 
  public:
   EnumeratorIterator(const Enumerator<T>* enumerator, HRESULT status)
-      : enumerator_(enumerator), status_(status), idx_(0), sz_(0) {}
```

Comment:
I noticed there was a bug here when doing my testing. The enumerator was previously always starting with an initial empty entry.